### DISCUSSION TITLE
QSelect auto select the first option?

### DISCUSSION THREAD

**Discussion by d-mok** [quasarframework/quasar#10183] [General - Components / Directives / etc] [answered]
Now, the QSelect component remains empty (i.e. unselected) when `options` is populated. I want it to select the first option automatically. And when the `options` array changes, it should:
1. if the currently selected item is also in the new `options`, it should kept being selected.
2. if not, then the first option in the new `options` should be selected.  

Is there a built-in attribute for this?

**Comment by ontwikkelfabriek** [comment, marked as answer]
@d-mok 

> Is there a built-in attribute for this?

No

Here's a codepen that **auto** selects the **first** option. Your other rules (1 and 2)  you'll have to try to code it yourself first:
https://codepen.io/ontwikkelfabriek/pen/ZEKvWmE?
